tests/typehints/test_script_decorator.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from subprocess import run
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+from typing import Iterator, Tuple
+
+import pytest
+
+from hera.workflows import Step, Task
+
+SIMPLE_SCRIPT = """
+from hera.workflows import script
+
+@script()
+def simple_script(input: str) -> int:
+    print(input)
+    return len(input)
+""".strip()
+
+
+def run_mypy(python_code: str):
+    with TemporaryDirectory() as d:
+        python_file = Path(d) / "example.py"
+        python_file.write_text(python_code)
+        mypy_cmd = ["mypy", "--config-file", "tests/typehints/test-mypy.toml", str(python_file)]
```

Comment:
These tests are a little bit slow (a couple of seconds each?) because they're invoking mypy here. We could combine them into fewer tests (e.g. one test that passes every Step parameter at once instead of one test per parameter), at the cost of making test failures more complex to diagnose